src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,154 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { postCommentData } from '../api/api';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  postId: number,
+  setPostComments: React.Dispatch<React.SetStateAction<Comment[]>>,
+  setErrorMessage: React.Dispatch<React.SetStateAction<boolean>>,
+};
+
+type ObjectFields = {
+  [key: string]: string,
+};
+
+type ObjectErrors = {
+  [key: string]: boolean,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  setPostComments = () => {},
+  setErrorMessage = () => {},
+}) => {
+  const [loader, setLoader] = useState<boolean>(false);
+
+  const [formComment, setFormComment] = useState<ObjectFields>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [formErrors, setFormErrors] = useState<ObjectErrors>({
+    nameError: false,
+    emailError: false,
+    bodyError: false,
+  });
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>
+  | React.ChangeEvent<HTMLTextAreaElement>) => {
+    const { name, value } = event.target;
+
+    setFormComment(prev => ({
+      ...prev,
+      [name]: value,
+    }));
+
+    setFormErrors(prev => ({
+      ...prev,
+      [`${name}Error`]: false,
+    }));
+  };
+
+  const resetFormClear = () => {
+    setFormComment({
+      name: '',
+      email: '',
+      body: '',
+    });
+
+    setFormErrors({
+      nameError: false,
+      emailError: false,
+      bodyError: false,
+    });
+  };
+
+  const resetFormAdd = () => {
+    setFormComment(prev => ({
+      ...prev,
+      name: '',
+      email: '',
+    }));
+
+    setFormErrors(prev => ({
+      ...prev,
+      nameError: false,
+      emailError: false,
+    }));
+  };
+
+  const addNewComment = async () => {
+    setLoader(true);
+
+    try {
+      const { name, email, body } = formComment;
+
+      const newComment = {
+        postId,
+        name: name.trim(),
+        email: email.trim(),
+        body: body.trim(),
+      };
+
+      const serverComment = await postCommentData(newComment);
+
+      setPostComments(prevComments => [...prevComments, serverComment]);
+
+      resetFormAdd();
+    } catch {
+      setErrorMessage(true);
+    } finally {
+      setLoader(false);
+    }
+  };
+
+  const handleAddComment = (event: React.FormEvent<HTMLFormElement>
+  | React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault();
+
+    let error = false;
+
+    if (!formComment.name.trim()) {
+      setFormErrors(prev => ({
+        ...prev,
+        nameError: true,
+      }));
+
+      error = true;
+    }
+
+    if (!formComment.email.trim()) {
+      setFormErrors(prev => ({
+        ...prev,
+        emailError: true,
+      }));
+
+      error = true;
+    }
+
+    if (!formComment.body.trim()) {
+      setFormErrors(prev => ({
+        ...prev,
+        bodyError: true,
+      }));
+
+      error = true;
+    }
+
+    if (error) {
+      return;
+    }
+
+    addNewComment();
+  };
```

Comment:
come on, it's easier to do early return instead
```suggestion
  const handleAddComment = (event: React.FormEvent<HTMLFormElement>
  | React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    event.preventDefault();

    if (!formComment.name.trim()) {
      setFormErrors(prev => ({
        ...prev,
        nameError: true,
      }));
      
      return;
    }

    if (!formComment.email.trim()) {
      setFormErrors(prev => ({
        ...prev,
        emailError: true,
      }));
      
      return;
    }

    if (!formComment.body.trim()) {
      setFormErrors(prev => ({
        ...prev,
        bodyError: true,
      }));
      
      return;
    }
    
    // if everything ok - add comment
    addNewComment();
  };
```